### PR TITLE
Completed Part 8.

### PR DESCRIPTION
**Pull request by pvardanis**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -89,15 +97,10 @@ struct Numeric
...
-#include <cmath>
-#include <functional>
-#include <memory>
-
-template<typename numType>
+template<typename NumericType>
 struct Numeric
 {
-    using myType = numType;
+    using myType = Temporary<NumericType>; //
```
Comment (matkatmusic) [inline main.cpp]: - [ ] `using` directives follow the same rules as Type Names regarding capitalization.

File: main.cpp [hunk 2]
```diff
@@ -226,17 +178,18 @@ struct Numeric<double>
                 std::cout << "Dividing by 0!" << std::endl;
             }
         }
-        
-        *value /= myNumber; 
+
```
Comment (matkatmusic) [inline main.cpp:191]: I have recently updated these instructions in Part7 accordingly: 
```
#11) now that your class is templated, you'll need to adjust your logic in your division function to handle if your input is a zero or not, based on your templated type.  
        - look up how to use std::is_same<>::value on cppreference to determine the type of your template parameter.
        
        - look up how to use std::numeric_limits<>::epsilon() to determine if you're dividing by a floating point 0

        - in plain-english, you'll need to implement logic like this:
        if your template type is an int
                if your parameter's type is also an int
                        if your parameter is 0
                                don't do the division
                else if it's less than epsilon
                        dont do the divison
        else if it's less than epsilon
                warn about doing the division

        - to make these checks work during compilation, your if() statements will need to be 'constexpr':  if constexpr (expression)
```
- [ ] revise your logic here.

File: main.cpp [hunk 3]
```diff
@@ -248,6 +201,12 @@ void updateValue(std::unique_ptr<numType>& value)
     *value += 5;
 }
 
+template<typename numType>
+void cube(std::unique_ptr<numType>& value)
+{
+    *value = *value * *value * *value;
```
Comment (matkatmusic) [inline main.cpp:217]: lol needs more `*` :-P

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 4]
```diff
@@ -166,18 +166,32 @@ struct Numeric
     template<typename OtherType>
     Numeric& operator/=(const OtherType& myNumber)
     {
-        if (myNumber == 0.f)
+        if (std::is_same<NumericType, int>::value) // had to change this now, because implicitly converting
+        // types could affect value (and epsilon)
         {
-            if (std::is_same<myType, int>::value) 
+            if (std::is_same<OtherType, int>::value)
```
Comment (matkatmusic) [inline main.cpp]: - [ ] re-read the final instruction for this part and use constexpr.  it'll eliminate the static_cast on 174.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 4 above)
Comment (pvardanis) [reply main.cpp]: Still don't understand what `constexpr` actually does (except that static_cast is not needed anymore).

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**